src/forward_diff.jl

```diff
@@ -0,0 +1,45 @@
+
+real_rep(a::AbstractArray{DualT}) where {TAG, T, DualT<:ForwardDiff.Dual{TAG, T}} = reinterpret(reshape, T, a)
+
+# multiplication of dual vector/matrix by standard matrix from the left
+@inline function _matmul!(_C::AbstractVecOrMat{DualT}, A::AbstractMatrix, _B::AbstractVecOrMat{DualT},
+                          α=One(), β=Zero(), nthread::Nothing=nothing, MKN=nothing, contig_axis=nothing) where {DualT <: ForwardDiff.Dual}
+    B = real_rep(_B)
+    C = real_rep(_C)
+
+    @tturbo for n ∈ indices((C, B), 3), m ∈ indices((C, A), (2, 1)), l in indices((C, B), 1)
+        Cₗₘₙ = zero(eltype(C))
+        for k ∈ indices((A, B), 2)
+            Cₗₘₙ += A[m, k] * B[l, k, n]
+        end
+        C[l, m, n] = α * Cₗₘₙ + β * C[l, m, n]
+    end
+
+    _C
+end
+
+# multiplication of dual matrix by standard vector/matrix from the right
+@inline function _matmul!(_C::AbstractVecOrMat{DualT}, _A::AbstractMatrix{DualT}, B::AbstractVecOrMat,
+                          α=One(), β=Zero(), nthread::Nothing=nothing, MKN=nothing) where {TAG, T, DualT <: ForwardDiff.Dual{TAG, T}}
+    if all((ArrayInterface.is_dense(_C), ArrayInterface.is_column_major(_C),
+            ArrayInterface.is_dense(_C), ArrayInterface.is_column_major(_C)))
+        # we can avoid the reshape and call the standard method
+        A = reinterpret(T, _A)
+        C = reinterpret(T, _C)
+        _matmul!(C, A, B, α, β, nthread, MKN)
+    else
+        # we cannot use the standard method directly
+        A = real_rep(_A)
+        C = real_rep(_C)
+
```

Comment:
These lines aren't covered. Mind adding a test using a `view` or `Adjoint`/`Transpose` so that the check fails?

Reply:
Good point; done in 10479ce